src/audio/buffer.c

```diff
@@ -77,7 +77,9 @@ struct comp_buffer *buffer_new(struct sof_ipc_buffer *desc)
 		return NULL;
 	}
 
-	memcpy(&buffer->ipc_buffer, desc, sizeof(*desc));
+	if (memcpy_s(&buffer->ipc_buffer, sizeof(buffer->ipc_buffer),
```

Comment:
@dabekjakub looks like @lgirdwood comment provided om memcpy_s handling in #1100 it's also applicable in #1103

Reply:
I dont't think this applies since return type is struct and not an error code. @lgirdwood  ?

Reply:
memcpy_s returns an int so it can be checked like
```
err = memcpy_s();
if (err)
    return NULL;
```
but the function returns a buffer * so you could return NULL in any error (and emit some trace, callers should also check.).

Ditto for the other if(memcpy_s) below too.

Reply:
Actualy I aligned to already existing code . This whole function uses return null as error handling. I don't want to include refactoring into already bloated commit.

Reply:
The already existing code probably slipped through and needs to be updated.  We do need to see error trace for the failures too since these memcpys are used by code that accept variable runtime arguments based on component/pipeline status.

Reply:
I agree it is fine idea to refactor whole thing, however we have critical bugs in queue that I need to take care of as soon as this change is merged and refactoring seems like a low priority task right now. Please set refactor as a enchantment since it should be done for the whole solution.

I will add trace messages in appropriate places in the commit thou.

Reply:
Please use the coding standard here, this means we have to fix less code later on.

Reply:
Fixed